src/rad/inspector/wxfbadvprops.cpp

```diff
@@ -578,10 +594,33 @@ wxVariant wxFBBitmapProperty::ChildChanged(wxVariant& thisValue, const int child
 
                     break;
                 }
-                // 'Load From Art Provider'
-                case 4:
+				// 'Load From XRC'
+				case 4:
+				{
+					if (prevSrc != 4)
+					{
+						for (unsigned int i = 1; i<count; i++)
+						{
+							wxPGProperty *p = Item(i);
+							if (p)
+							{
+								wxLogDebug(wxT("wxFBBP::ChildChanged: Removing:%s"), p->GetLabel().c_str());
+								GetGrid()->DeleteProperty(p);
+							}
+						}
+						bp->AppendChild(bp->CreatePropertyXrcName());
+					}
+
+					if (childVals.GetCount() == 2)
+						newVal = childVals.Item(0) + wxT("; ") + childVals.Item(1);
+					else if (childVals.GetCount() > 1)
+						newVal = childVals.Item(0) + wxT("; ");
```

Comment:
Hm ... why is `== 2` a special case?

Reply:
Because copy-paste. Each case uses this code fragment, some check for `== 2`, some for `== 3`, doesn't make sense for the `== 2` case, but looks uniform.

Reply:
I see. What happens when `childVals.GetCount() == 1`?

Reply:
Good question. Currently this can't happen because each case adds at least one child so there are always at least 2 child elements, but otherwise it looks like the previous value gets kept.